request/util.go

```diff
@@ -0,0 +1,24 @@
+package request
+
+import (
+	"github.com/ONSdigital/go-ns/log"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+// DrainBody drains the body of the given of the given HTTP request.
+// any handler function that reads the request body should ensure that this
+// function is called.
+// ``
+func DrainBody(r *http.Request) {
+	_, err := io.Copy(ioutil.Discard, r.Body)
+	if err != nil {
+		log.InfoCtx(r.Context(), "error draining request body", nil)
+	}
+
+	err = r.Body.Close()
+	if err != nil {
+		log.InfoCtx(r.Context(), "error closing request body", nil)
+	}
```

Comment:
why not return errors?

Reply:
hmm, I see the caller is in a `defer` - which answers that question

Reply:
I was unsure whether to return errors from this function. I realise that typically errors are not returned from lib's, though this seemed to sit somewhere between. I wrote the function and then decided to put it in go-ns for reuse. I could return errors and have them handled in a deferred function, though it would make the code more verbose. What do you think?

Reply:
It's likely to be in a `defer` and we are discarding the body, so I think your first instinct - to log and ignore - is best